.github/workflows/ci-lambda.yml

```diff
@@ -0,0 +1,98 @@
+## [
+##  { projectName: "sftp-outbound-transfer-lambda", projectRoot: "ops/terraform/services/eft/lambda_src/sftp_outbound_transfer", testsRoot: "ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/tests", pythonVersion: "3.12" },
+##  { projectName: "update-pipeline-slis-lambda", projectRoot: "ops/terraform/services/pipeline/modules/bfd_pipeline_slis/lambda_src", testsRoot: "ops/terraform/services/pipeline/modules/bfd_pipeline_slis/lambda_src/tests", pythonVersion: "3.11" }
+## ]
+
+jobs:
+  python-check:
+    runs-on: ubuntu-latest
+    outputs:
+      jobstatus: ${{ join(steps.*.conclusion) }}
+      MATRIX: ${{ steps.match-proj-deltas.outputs.MATRIX }}
+
+    steps:
+      - name: Checkout code
+        id: code-checkout
+        uses: actions/checkout@v4
+        with:
+            fetch-depth: 2
+
+      - name: Get changed directory list as CSV with count
+        id: changed-dirs
+        uses: tj-actions/changed-files@v44
+        with:
+          dir_names: "true"
+          separator: ","
+              
+      - name: Capture Modified Folders with Count
+        id: validate-testworthy-changes
+        run: |
+            UNIQ_FOLDERS=${{ runner.temp }}/UNIQFLDR.txt
+            echo ${{ steps.changed-dirs.outputs.all_changed_files }} >> ${UNIQ_FOLDERS}
+            UNIQ64=$(cat ${UNIQ_FOLDERS}|base64)
+            echo "FOLDERLISTCSV=${UNIQ64}" >> $GITHUB_OUTPUT
+            echo "FOLDERCT=${{ steps.changed-dirs.outputs.all_changed_files_count }}"
+
+      - name: Define Managed Project Roots
+        id: define-proj-list
+        run: |
+            PROJCFG_FN=${{ runner.temp }}/${{ env.PRJCONFIGFN }}
+            echo '[ { "projectName": "sftp-outbound-transfer-lambda", "projectRoot": "ops/terraform/services/eft/lambda_src/sftp_outbound_transfer", "testsRoot": "ops/terraform/services/eft/lambda_src/sftp_outbound_transfer/tests", "pythonVersion": "3.12" }, { "projectName": "update-pipeline-slis-lambda", "projectRoot": "ops/terraform/services/pipeline/modules/bfd_pipeline_slis/lambda_src", "testsRoot": "ops/terraform/services/pipeline/modules/bfd_pipeline_slis/lambda_src/tests", "pythonVersion": "3.11" } ]' > ${PROJCFG_FN}
+
+      - name: Match Changed Folder List to Project Roots
+        id: match-proj-deltas
+        run: |
+            PROJ_JSON="${{ runner.temp }}/${{ env.PRJCONFIGFN }}"
+            PROJ_ROOT_LIST=$(cat ${PROJ_JSON} | jq -c '[ .[]|.projectRoot ]')
+            FOLDERLISTCSV64=${{ steps.validate-testworthy-changes.outputs.FOLDERLISTCSV }}
+            FOLDERLISTJSON=$(echo ${FOLDERLISTCSV64}|base64 -d| awk '{ n=split($0,a,","); { for(i in a) printf "%s\n",a[i]; } }'|sort|uniq| jq --raw-input '.' | jq --slurp)
+            PROJECTMATCHJSON=$(echo ${FOLDERLISTJSON} | jq '.[]'|sed 's/"//g'|while read pathnm; do cat ${PROJ_JSON}| jq --arg match $pathnm '.[]|select(.projectRoot|startswith($match))'; done | jq '[.]')
+            echo "MATRIX="$PROJECTMATCHJSON >> $GITHUB_OUTPUT
```

Comment:
nit: This seems quite complicated; maybe it could be replaced by an action like [Conditional Build Matrix]https://github.com/marketplace/actions/conditional-build-matrix)? Generating a [JMESPath](https://jmespath.org/) for checking if any of the matrix JSON contains the changed paths should be a bit easier than doing the checks manually here.

It'll also make it easier for someone to add new projects to the input JSON as it would be split into another file.

Nit because this looks like it's working, and I know that there's already been a significant amount of effort put into this.

Reply:
the match-proj-deltas step is actually about as simple as possible, all things considered.

Reply:
Resolving this conversation as we've decided to forgo the complexity involved with filtering the matrix and instead just use a static matrix to run the tests whenever any python file or requirements files change.